datalad/crawler/nodes/tests/test_misc.py

```diff
@@ -47,6 +48,29 @@ def test_get_disposition_filename():
     eq_(output[0]['filename'], 'T1.nii.gz')
 
 
+def test_sink():
+    data = {'x': 'y', 'g': 'h', 'a': 'b'}
+
+    keys = ['x', 'a']
+    nomatch = ['z']
+
+    # no arguments
+    genempty = Sink()
+    eq_(list(genempty(data)), [{'a': 'b', 'x': 'y', 'g': 'h'}])
+
+    # if key for sunk data is specified
+    gen = Sink(output='result')
+    eq_(list(gen(data)), [{'a': 'b', 'x': 'y', 'result': [{'a': 'b', 'x': 'y', 'g': 'h'}], 'g': 'h'}])
+
+    # if list of keys is specified
+    genkeys = Sink(keys, 'result')
+    eq_(list(genkeys(data)), [{'a': 'b', 'x': 'y', 'result': [{'a': 'b', 'x': 'y'}], 'g': 'h'}])
+
+    # if list of keys has no match
+    gentwo = Sink(nomatch, 'result')
+    eq_(list(gentwo(data)), [{'a': 'b', 'result': [{}], 'x': 'y', 'g': 'h'}])
```

Comment:
oh -- one more test please -- so if you call the same sink second time, it does append new data to the previously collected one.  Also check that actual .data of the sink does contain those entries (above tests tested only output of the call but not the internal state, i.e. `.data` of the sink).